ap/columnar_util.py

```diff
@@ -827,9 +846,10 @@ def __init__(
 
         # store attributes
         self.shifts = set(law.util.make_list(shifts)) if shifts else set()
+        self.update_func = None
         self.requires_func = None
         self.setup_func = None
-        self.producer_kwargs = None
+        self.call_kwargs = None
 
     @property
     def all_shifts(self):
```

Comment:
Also propagate to `self.shifts` (see comment for `run_update`)

Reply:
✔︎

Reply:
👍